python/lsst/ip/isr/isrFunctions.py

```diff
@@ -292,31 +291,57 @@ def illuminationCorrection(maskedImage, illumMaskedImage, illumScale):
 
 def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1, collapseRej=3.0,
                        statControl=None):
-    """Apply overscan correction in place
-
-    @param[in,out] ampMaskedImage  masked image to correct
-    @param[in] overscanImage  overscan data as an afw.image.Image or afw.image.MaskedImage.
-                              If a masked image is passed in the mask plane will be used
-                              to constrain the fit of the bias level.
-    @param[in] fitType  type of fit for overscan correction; one of:
-                        - 'MEAN'
-                        - 'MEDIAN'
-                        - 'POLY' (ordinary polynomial)
-                        - 'CHEB' (Chebyshev polynomial)
-                        - 'LEG' (Legendre polynomial)
-                        - 'NATURAL_SPLINE', 'CUBIC_SPLINE', 'AKIMA_SPLINE' (splines)
-    @param[in] order  polynomial order or spline knots (ignored unless fitType
-                      indicates a polynomial or spline)
-    @param[in] collapseRej  Rejection threshold (sigma) for collapsing dimension of overscan
-    @param[in] statControl  Statistics control object
+    """Apply overscan correction in-place
+
+    The ``ampMaskedImage`` and ``overscanImage`` are modified, with the fit
+    subtracted. Note that the ``overscanImage`` should not be a subimage of
+    the ``ampMaskedImage``, to avoid being subtracted twice.
```

Comment:
This is a good point. Is it possible to add a test to check that `overscanImage` is not a subimage of `ampMaskedImage`? You say in your commit message that it's not clear how to detect this case, but I think it's rather important if it's doable.

Reply:
I don't believe it's possible to test, since we have no way of determining whether two images share the same physical pixels. I therefore consider that the only way to deal with this is to document the possibility.